### ISSUE TITLE
Accessibility: aria-hidden textarea is focusable via keyboard

### ISSUE THREAD

**Issue by Rosita311** [pluginkollektiv/antispam-bee#696]
### Describe the bug
The honeypot textarea has aria-hidden="true" but no tabindex="-1", which means keyboard users can still focus it. This violates the WCAG rule that aria-hidden elements must not be focusable, and triggers the following axe DevTools warning: "Ensure aria-hidden elements are not focusable nor contain focusable elements".

### To Reproduce
Install Antispam Bee on a WordPress site with comments enabled
Open a post with a comment form
Navigate through the form using the Tab key
The honeypot textarea receives keyboard focus despite having aria-hidden="true"
Expected behavior
The honeypot textarea should have tabindex="-1" so keyboard users cannot focus it. The honeypot functionality remains intact.

```
<!-- Current -->
<textarea aria-hidden="true" name="comment" ...></textarea>

<!-- Expected -->
<textarea aria-hidden="true" tabindex="-1" name="comment" ...></textarea>
```
### System:
WordPress: 7.0
Antispam Bee: 2.11.12

### Additional context
Reference: [aria-hidden-focus — Deque University](https://dequeuniversity.com/rules/axe/4.4/aria-hidden-focus)

I applied a JavaScript workaround in my theme as a temporary fix:

```
document.querySelectorAll('[aria-hidden="true"]').forEach(el => {
  if (['A', 'BUTTON', 'INPUT', 'TEXTAREA', 'SELECT'].includes(el.tagName)) {
    el.setAttribute('tabindex', '-1');
  }
  el.querySelectorAll('a, button, input, textarea, select, [tabindex]').forEach(focusable => {
    focusable.setAttribute('tabindex', '-1');
  });
});
```
The fix should ideally be in the plugin itself rather than requiring a theme-level workaround.

**Comment by Zodiac1978** [comment]
I am not sure why this happens for you, but we have `tabindex="-1"` on the honeypot for 9 years already:
https://github.com/pluginkollektiv/antispam-bee/commit/0864a85a89262b52f139ff2fbf03cd101e720877

If you share the link where this happens, we can have a closer look why it happens.

**Comment by Rosita311** [comment]
Thanks for your reaction. I'll remove the JS so you can check it out. 

Here's a link to a blog, with a commentform: https://www.rulesbyrosita.nl/een-technisch-kijkje-in-mijn-wordpress-thema/

Maybe this has something to do with another plugin (or my theme), I'll check that out.

**Comment by Zodiac1978** [comment]
I'm at WordCamp Europe right now, and I don't have my laptop with me. I will have a look on my train ride back tomorrow.

**Comment by Rosita311** [comment]
Nice, have fun! And take your time. 

I commented out the Javascript, so you can check the issue. I am going to check for other conflicts. I was thinking about a setting with tabindex in WP Accessibility.

**Comment by Zodiac1978** [comment]
Hi @Rosita311 

I took a look at your website and found the problem.

You are using WP Accessibility from @joedolson and there is this feature in it:

> Remove tabindex from elements that are focusable. (Also fixes plugin-caused problems.)

I blocked the `wp-accessibility.min.js` and the problem is gone. Re-allowed it and it is re-appearing.

So you are absolutely right with your assumption.

Not sure how to fix this, but I hope Joe can help here.

//cc: @rianrietveld

**Comment by rianrietveld** [comment]
@joedolson, this issue may be of interest for you.

**Comment by joedolson** [comment]
I added a commit against [the parallel issue in the WP Accessibility repo](https://github.com/joedolson/wp-accessibility/issues/56) which would allow you to add an ignore class on the element. If that's not something you'd want to do, let me know; I can consider other options, too.

**Comment by joedolson** [comment]
@Rosita311 I'll also note that it may be worthwhile for you to check whether WP Accessibility is actually removing any tabindex attributes that are actually problems. If it's not, then you could just turn that feature off in the admin.

You can tell what WP Accessibility is doing by turning on stats or by viewing the report of activity in the browser console when browsing the site as an admin.

**Comment by Zodiac1978** [comment]
> If it's not, then you could just turn that feature off in the admin.

Yes, I think this would be the best approach for your site @Rosita311 

For the general problem, I have commented on the other repo.

**Comment by Rosita311** [comment]
Thank you all so much for your help! @Zodiac1978 @joedolson and @rianrietveld I think it's the right approach to turn this setting off as well. I was tinkering with the settings of WP accessibility, now that I have my own theme, I was thinking about what settings are needed and what settings are not needed anymore.

**Comment by Zodiac1978** [comment]
This is fixed upstream in https://github.com/joedolson/wp-accessibility/issues/56 / https://github.com/joedolson/wp-accessibility/commit/64d15da075fa4fcde22c977dca2b8d3044de7655
